src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php

```diff
@@ -66,6 +66,46 @@ public function register_routes() {
 	 * @return bool True if the request has read access for the item, otherwise false.
 	 */
 	public function get_item_permissions_check( $request ) {
+		$options = $this->get_registered_options();
+
+		foreach ( $options as $name => $args ) {
+			$test = call_user_func_array( $args['read_permission_callback'], array( $name ) );
+			if ( $test ) {
+				return true;
+			}
+		}
+
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Checks if a request has access to update the specified settings.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to update the item, WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		$options = $this->get_registered_options();
+
+		foreach ( $options as $name => $args ) {
+			if ( isset( $request[ $name ] ) ) {
+				$has_permissions = call_user_func_array( $args['edit_permission_callback'], array( $name ) );
+				if ( ! $has_permissions ) {
+					return new WP_Error(
+						'rest_cannot_update',
+						/* translators: %s: Custom field key. */
+						sprintf( __( 'Sorry, you are not allowed to edit the %s option.' ), $name ),
+						array(
+							'key'    => $name,
+							'status' => rest_authorization_required_code(),
+						)
+					);
+				}
```

Comment:
The `WP_Test_REST_Settings_Controller::test_update_item_with_edit_permission_callbacks()` test method implies that the setting can be updated even if the current user lacks the `manage_options` capability. However, the test fails because the code still checks for that capability.
I'm not sure what the intention is here: should a user be able to update a setting even without the `manage_options` capability, as long as the permission callback allows it?
```suggestion
				}
				return true;
```